src/main/scripts/cdp-listend

```diff
@@ -192,25 +192,25 @@ while($ip1->recv($newmsg,1024)) {
     my $ntype = $1;
     my $time  = $2;
     logit('info',"Received UDP packet ($ntype|$time) from $r_host");
-    # time extraction!
-    if ($ntype eq"ccm") {
-        my $smear = int(rand($fetch_smear));
-        my $delay = $smear + $fetch_offset;
-        logit('info', "$fetch will be called in $delay seconds (offset=$fetch_offset, smear=$smear)");
-        sleep($delay);
-        logit('info', "Calling $fetch with unix time $time (after $delay seconds)");
-        system ("$fetch --profile-time=$time") == 0
-            or logit('err',"[ERROR] call of ccm-fetch ($fetch) failed: $!");
+    if ($ntype eq "ccm") {
+        my $delay = int(rand($fetch_smear)) + $fetch_offset;
+        call_handler($fetch, "--profile-time=$time", $delay);
     } elsif ($ntype eq "cdb") {
-        my $smear = int(rand($nch_smear));
-        logit('info', "$nch will be called in $smear seconds");
-        sleep($smear);
-        logit('info', "Calling $nch with unix time $time (after $smear seconds)");
-        system ("$nch") == 0
-            or logit('err',"[ERROR] call of cdbsync-nch ($nch) failed: $!");
+        my $delay = int(rand($nch_smear));
+        call_handler($nch, "", $delay);
+    } else {
+        logit('err', "[ERROR] Ignoring packet, unknown notification type");
     }
 }
 
+sub call_handler {
+    my ($handler, $handler_args, $delay) = @_;
+    logit('info', "$handler will be called in $delay seconds");
+    sleep($delay);
+    logit('info', "Calling $handler now");
```

Comment:
Just realised this drops the $time from the log message; does something else log the time of this invocation or are we losing the info about when the handler was actually called?

Reply:
It is still logged once when the notification packet is received, but as it is the embedded timestamp from the packet it isn't that helpful IMHO, e.g:

Previous behaviour:
```
Sep  8 00:02:02 example-host /usr/sbin/cdp-listend[4836]: Received UDP packet (ccm|1725750122) from 198.51.18.11
Sep  8 00:02:02 example-host /usr/sbin/cdp-listend[4836]: /usr/sbin/ccm-fetch will be called in 0 seconds (offset=0, smear=0)
Sep  8 00:02:02 example-host /usr/sbin/cdp-listend[4836]: Calling /usr/sbin/ccm-fetch with unix time 1725750122 (after 0 seconds)
```

New behaviour:
```
Sep  9 10:48:14 example-host /usr/sbin/cdp-listend[29011]: Received UDP packet (ccm|1725875294) from 198.51.18.11
Sep  9 10:48:14 example-host /usr/sbin/cdp-listend[29011]: /usr/sbin/ccm-fetch will be called in 0 seconds
Sep  9 10:48:14 example-host /usr/sbin/cdp-listend[29011]: Calling /usr/sbin/ccm-fetch now
```